core/build/devenv.odin

```diff
@@ -0,0 +1,199 @@
+package build
+
+import "core:unicode/utf8"
+import "core:runtime"
+import "core:os"
+import "core:fmt"
+import "core:strings"
+import "core:encoding/json"
+import "core:path/filepath"
+
+// Important Note(Dragos): We need a system for handling imported dependencies when configuring the devenv because the paths might be goofy. This might also impact odin build... We'll see...
+
+VSCode_Debugger_Type :: enum {
+	cppvsdbg,
+	cppdbg,
+}
+
+VSCode_Request_Type :: enum {
+	launch,
+}
+
+VSCode_Config_Json :: struct {
+	type: string,
+	request: string,
+	preLaunchTask: string,
+	name: string,
+	program: string,
+	args: [dynamic]string,
+	cwd: string,
+}
+
+VSCode_Launch_Json :: struct {
+	version: string,
+	configurations: [dynamic]VSCode_Config_Json,
+}
+
+VSCode_Task_Json :: struct {
+	label: string,
+	type: string,
+	command: string,
+}
+
+VSCode_Tasks_Json :: struct {
+	version: string,
+	tasks: [dynamic]VSCode_Task_Json,
+}
+
+_generate_vscode :: proc(target: ^Target, settings: Settings, opts: Dev_Options) {
+	config := target_config(target, settings) // Note(Dragos): I really don't like settings being based around
+
+	if os.is_dir(".vscode") || os.make_directory(".vscode") == os.ERROR_NONE {
+		launch_file, launch_err := os.open(".vscode/launch.json", os.O_CREATE | os.O_TRUNC)
+		tasks_file, tasks_err := os.open(".vscode/tasks.json", os.O_CREATE | os.O_TRUNC)
+		if launch_err == os.ERROR_NONE && tasks_err  == os.ERROR_NONE {
+			defer {
+				os.close(launch_file)
+				os.close(tasks_file)
+			}
+			launch_json: VSCode_Launch_Json
+			
+			tasks_json: VSCode_Tasks_Json
+			tasks_json.version = "2.0.0"
+			launch_json.version = "0.2.0"
+			
+			debug_config_json: VSCode_Config_Json
+			if opts.custom_cwd != "" {
+				fmt.eprintf("Cannot currently have a custom cwd for vscode.\n")
+			} else if .Cwd_Out in opts.flags {
+				debug_config_json.cwd = filepath.join({"%{workspaceFolder}/", config.out_dir}) // Note(Dragos): Test this
```

Comment:
Why is this one written with `%` sign instead of `$`?

Reply:
Might be a typo, but i think thats the vscode way of doing it. Its for vscode

Reply:
Yes, good catch. Typo.